test/grdfilter/filtersample.sh

```diff
@@ -1,7 +1,7 @@
 #!/usr/bin/env bash
 ps=filtersample.ps
 
-if [[ ${HAVE_GLIB_GTHREAD} =~ TRUE|ON ]]; then
+if [ "X$(gmt-config --has-gthreads)" = "Xyes" ]; then
```

Comment:
We have had a mix of TRUE|ON vs gmt-config.  The TRUE|ON does not work for me.  Perhaps we need a short shell plug-n that checks OS and applies these under if [ $OS =....]?

Reply:
So some checking shows that it is only the Glib thread cases that use -a:

/grdfilter/threads.sh:	_thread_opt=-x+a
./grdfilter/filtertest.sh:  _thread_opt=-x+a
./grdfilter/spikes.sh:  _thread_opt=-x+a
./grdfilter/filtersample.sh:  _thread_opt=-x+a

All OpenMP modules compiled with gcc-mp automatically use all cores available unless -x1 is set, so presumably those have all worked fine.  But looks like we need to improve the checking for theGlib threads.
ANother issue: Why does your glib parsing deal with a modifier +a that is the default for OpenMP parser?  If OPenMP is build in then the default (in say grdmath) is what you call
l -x+a.  Dont you want the glib threads to be the default is they are available, so we could standardise on one rst file for -x and one synopsis string for -x? Here is from grdfilter:

<img width="761" alt="xg" src="https://github.com/GenericMappingTools/gmt/assets/26473567/7e89cae3-dfa9-4901-9f86-002db1d6dd1d">
 
See below for GMT standard:

<img width="608" alt="x" src="https://github.com/GenericMappingTools/gmt/assets/26473567/58f92eea-2573-4504-9413-eac08fe717cd">

GMT shows no modifier +a. Can we fix this so we can clean up those 4 scripts (no more checking since the module knows and with elimination of +a will automatically run in parallel and no kludge in the scrips are needed; similar to OpenMP)

Reply:
I guess -x+a predates that unified OMP -x

But let's release 6.5 otherwise these last changes never finish.